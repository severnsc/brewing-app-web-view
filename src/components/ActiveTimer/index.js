import React from "react"
import styles from "./styles"
import PropTypes from "prop-types"

const ActiveTimer = ({ name, time, isRunning, startTimer, stopTimer, resetTimer }) => {

	const startButton = <button onClick={startTimer} style={styles.start}>Start</button>
	const stopButton = <button onClick={stopTimer} style={styles.stop}>Stop</button>

	return (
		<div>
			<h2>{name}</h2>
			<h2>{time}</h2>
			{isRunning ? stopButton : startButton}
			<button onClick={resetTimer} style={styles.button}>Reset</button>
		</div>
	)
}

ActiveTimer.propTypes = {
	name: PropTypes.string.isRequired,
	time: PropTypes.string.isRequired,
	isRunning: PropTypes.bool.isRequired,
	startTimer: PropTypes.func.isRequired,
	stopTimer: PropTypes.func.isRequired,
	resetTimer: PropTypes.func.isRequired
}

export default ActiveTimer