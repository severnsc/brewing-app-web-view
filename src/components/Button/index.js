import React from "react"
import PropTypes from "prop-types"
import globalStyles from "../styles"

const Button = ({ onClick, value, style }) => (
	<button style={{...globalStyles.button, ...style}} onClick={onClick}>
		{value}
	</button>
)

Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	value: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number
	]).isRequired,
	style: PropTypes.object
}

export default Button