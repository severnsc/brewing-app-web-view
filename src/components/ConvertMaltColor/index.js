import React from "react"
import PropTypes from "prop-types"
import { MaltColor } from ".."
import { convertColor } from "../../utils"

const ConvertMaltColor = ({ from, to, value }) => {

	const convertedColor = convertColor(from, to, value)

	let srmValue
	if(from === "SRM"){
		srmValue = value
	}else{
		srmValue = convertColor(from, "SRM", value)
	}

	return <MaltColor srmValue={srmValue} value={convertedColor} />

}

ConvertMaltColor.propTypes = {
	from: PropTypes.oneOf(["SRM", "L", "EBC"]).isRequired,
	to: PropTypes.oneOf(["SRM", "L", "EBC"]).isRequired,
	value: PropTypes.number.isRequired
}

export default ConvertMaltColor