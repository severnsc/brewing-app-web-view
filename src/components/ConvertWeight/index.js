import React from "react"
import PropTypes from "prop-types"
import { Weight } from ".."
import { convertLbsToKg, convertKgToLbs } from "../../utils"

const ConvertWeight = ({ from, to, amount }) => {

	let convertedAmount
	if(from === "imperial" && to === "metric"){
		convertedAmount = convertLbsToKg(amount)
	}

	if(from === "metric" && to === "imperial"){
		convertedAmount = convertKgToLbs(amount)
	}

	return(
		<Weight unit={to} amount={convertedAmount} />
	)
}

ConvertWeight.propTypes = {
	from: PropTypes.oneOf(["metric", "imperial"]).isRequired,
	to: PropTypes.oneOf(["metric", "imperial"]).isRequired,
	amount: PropTypes.number.isRequired
}

export default ConvertWeight