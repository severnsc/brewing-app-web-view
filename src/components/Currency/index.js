import React from "react"
import PropTypes from "prop-types"

const Currency = ({ amount, unit }) => {

	let currencySymbol
	switch(unit){
		case "USD":
			currencySymbol = "$"
			break

		case "GBP":
			currencySymbol = "£"
			break

		case "EUR":
			currencySymbol = "€"
			break

		default:
			currencySymbol = ""
	}

	const parsedAmount = parseFloat(amount).toFixed(2)

	return(
		<span>
			{currencySymbol}{parsedAmount}
		</span>
	)
}

Currency.propTypes = {
	amount: PropTypes.number.isRequired,
	unit: PropTypes.oneOf(["USD", "GBP", "EUR"]).isRequired
}

export default Currency