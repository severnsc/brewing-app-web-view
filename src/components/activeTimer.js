import React from "react"
import PropTypes from "prop-types"

const ActiveTimer = ({ time, isRunning, startTimer, stopTimer, resetTimer }) => {

	const styles = {
		button: {
			color: "white",
			backgroundColor: "grey"
		}
	}

	styles.start = {
		...styles.button,
		backgroundColor: "green"
	}

	styles.stop = {
		...styles.button,
		backgroundColor: "red"
	}

	const startButton = <button onClick={startTimer} style={styles.start}>Start</button>
	const stopButton = <button onClick={stopTimer} style={styles.stop}>Stop</button>

	return (
		<div>
			<h2>{time}</h2>
			{isRunning ? stopButton : startButton}
			<button onClick={resetTimer} style={styles.button}>Reset</button>
		</div>
	)
}

ActiveTimer.propTypes = {
	time: PropTypes.string.isRequired,
	isRunning: PropTypes.bool.isRequired,
	startTimer: PropTypes.func.isRequired,
	stopTimer: PropTypes.func.isRequired,
	resetTimer: PropTypes.func.isRequired
}

export default ActiveTimer