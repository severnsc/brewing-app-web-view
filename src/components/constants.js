export const errorRed = "hsl(353, 85%, 44%)"
export const green = "hsl(167, 90%, 38%)"