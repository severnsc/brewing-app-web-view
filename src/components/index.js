import Timer from "./Timer"
import ForgotPasswordForm from "./ForgotPasswordForm"
import HoverableTableRow from "./HoverableTableRow"
import InventoryItemForm from "./InventoryItemForm"
import Modal from "./Modal"
import NavBar from "./NavBar"
import ResetPasswordForm from "./ResetPasswordForm"
import SearchBar from "./SearchBar"
import SortableTable from "./SortableTable"
import TimerAlertForm from "./TimerAlertForm"
import TimerForm from "./TimerForm"
import InventoryForm from "./InventoryForm"
import FixedNewItemButton from "./FixedNewItemButton"
import NewTimerForm from "./NewTimerForm"
import NewTimerAlertForm from "./NewTimerAlertForm"
import {
	SignupForm,
	LoginForm
} from "./Forms"

export {
	Timer,
	ForgotPasswordForm,
	HoverableTableRow,
	InventoryItemForm,
	LoginForm,
	Modal,
	NavBar,
	ResetPasswordForm,
	SearchBar,
	SignupForm,
	SortableTable,
	TimerAlertForm,
	TimerForm,
	InventoryForm,
	FixedNewItemButton,
	NewTimerForm,
	NewTimerAlertForm
}