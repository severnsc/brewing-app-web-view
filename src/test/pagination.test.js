import React from 'react';
import renderer from "react-test-renderer"
import { Pagination } from "../components"
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("pagination", () => {

	it("renders a pagination component", () => {
		const pagination = renderer.create(<Pagination totalPages={2} page={0} />)
		const tree = pagination.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("does not render a previous button if page is 0", () => {
		const pagination = shallow(<Pagination totalPages={2} page={0} />)
		expect(pagination.findWhere(n => 
			n.text() === "< Previous" && n.type() === "span").length
		).toBe(0)
	})

	it("renders a span with text < Previous is page is > 0", () => {
		const pagination = shallow(<Pagination totalPages={2} page={1} />)
		expect(pagination.findWhere(n => 
			n.text() === "< Previous" && n.type() === "span").length
		).toBe(1)
	})

	it("renders a span with text Next > if page < totalPages", () => {
		const pagination = shallow(<Pagination totalPages={2} page={1} />)
		expect(pagination.findWhere(n => 
			n.text() === "Next >" && n.type() === "span").length
		).toBe(1)
	})

	it("does not render a next button if page === totalPages", () => {
		const pagination = shallow(<Pagination page={2} totalPages={2} />)
		expect(pagination.findWhere(n => 
			n.text() === "Next >" && n.type() === "span").length
		).toBe(0)
	})

})