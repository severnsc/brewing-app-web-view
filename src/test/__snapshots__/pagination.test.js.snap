// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination renders a pagination component 1`] = `
<div>
  <div>
    <span
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "padding": "0px 5px",
          "textDecoration": "underline",
          "width": "86.73px",
        }
      }
    >
      Next &gt;
    </span>
  </div>
</div>
`;
